### PR TITLE
Changed All Component Terra-Core Dependency Versions

### PR DESCRIPTION
**Pull request by amichaelparker**
### Summary
Changed the dependency versions of all terra-core components to match hoists to remove lerna warnings for those components.

Closes [#400](https://github.com/cerner/terra-framework/pull/400)

### CODE REVIEW

File: packages/terra-clinical-data-grid/package.json [hunk 1]
```diff
@@ -37,13 +37,13 @@
     "prop-types": "^15.5.8",
     "react-draggable": "^3.0.5",
     "resize-observer-polyfill": "^1.4.1",
-    "terra-button": "^2.0.1",
```
Comment (amichaelparker) [inline packages/terra-clinical-data-grid/package.json]: This leaves react-intl as the only lerna warning (2.7.0 vs 2.4.0).

Comment (emilyrohrbough) [reply packages/terra-clinical-data-grid/package.json]: Lets go ahead and match those as well. It looks like onset-picker is using 2.4.0. We can bump it to 2.7.0 to match data-grid or fall back to 2.0.0

Comment (amichaelparker) [reply packages/terra-clinical-data-grid/package.json]: It was set to 2.7.0, and falling back to 2.0.0 also gives a hoist warning. Setting it to 2.4.0 seems to work.

Comment (emilyrohrbough) [reply packages/terra-clinical-data-grid/package.json]: We would need to bump onset picker to 2.7.0 to match the data-grid's specification. I"m good with either way; the latest will be installed regardless.

Comment (amichaelparker) [reply packages/terra-clinical-data-grid/package.json]: I went ahead and set both onset picker and data grid to 2.7.0: [b34fb92](https://github.com/cerner/terra-clinical/pull/448/commits/b34fb9201be7da2bdc94c4c4733f804eb77a545d)

**Comment by yuderekyu** [comment]
sorry, we stealthily did most of this in the theme PR 😅

**Comment by amichaelparker** [comment]
It happens... Pretty regularly. :D It looks like data-grid and item-view are still having problems with the icon version. Setting it to ^2.0.0 doesn't resolve it anymore, but ^2.9.0 does, so now I'm testing that to check for issues.

File: packages/terra-clinical-data-grid/package.json [hunk 2]
```diff
@@ -37,13 +37,13 @@
...
     "terra-clinical-item-view": "^2.16.0",
-    "terra-content-container": "^2.0.0",
-    "terra-doc-template": "^1.0.0",
-    "terra-icon": "^2.0.0",
-    "terra-mixins": "^1.0.0",
-    "terra-overlay": "^2.0.0"
+    "terra-content-container": "^2.0.1",
+    "terra-doc-template": "^2.0.0",
+    "terra-icon": "^2.6.0",
+    "terra-mixins": "^1.21.0",
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/package.json]: uh oh, looks like we lost some of the changes you originally made

Comment (amichaelparker) [reply packages/terra-clinical-data-grid/package.json]: Should I change those back? The new changes from master don't throw the same hoist warnings.

Comment (emilyrohrbough) [reply packages/terra-clinical-data-grid/package.json]: yeah, we should match all dependencies to a major version `^2.0.0` or `^3.0.0` etc

Comment (emilyrohrbough) [reply packages/terra-clinical-data-grid/package.json]: it will make it easier to manage if/when we add new components to the repo

Comment (amichaelparker) [reply packages/terra-clinical-data-grid/package.json]: Packages updated!

File: packages/terra-clinical-detail-view/package.json [hunk 3]
```diff
@@ -27,11 +27,10 @@
   },
   "dependencies": {
     "classnames": "^2.2.5",
-    "prop-types": "^15.5.8",
```
[the remark this answers is not in the document]
Comment (amichaelparker) [reply packages/terra-clinical-detail-view/package.json]: Yeah, accidentally hacked that off while fixing merge conflicts.😀

[the remark this answers is not in the document]
Comment (amichaelparker) [reply packages/terra-clinical-detail-view/package.json]: I also forgot to bump one back down to the nearest major version, so I'm on a kind of roll this morning.

**Comment by emilyrohrbough** [comment]
A few changes: (I think it got lost in a merge)
- update `terra-doc-template` dependency to be `^2.0.0` [here](https://github.com/cerner/terra-clinical/blob/master/packages/terra-clinical-application/package.json#L36)
- update `terra-clinical-item-display` to be `^3.0.0` [here](https://github.com/cerner/terra-clinical/blob/master/packages/terra-clinical-item-view/package.json#L32) and [here](https://github.com/cerner/terra-clinical/blob/master/packages/terra-clinical-label-value-view/package.json#L32)

**Pull request merged**